### PR TITLE
Allow attaching MySky identities to accounts users

### PR DESCRIPTION
**Pull request by ro-tex**
# PULL REQUEST

## Overview

This PR allows for updating the user's pubKey via the `PUT /user` endpoint.

It depends on #83 and should be reviewed after that one is merged. The PR is currently pointing to that PR's branch, so the differences can be easily seen.

## Checklist
Review and complete the checklist to ensure that the PR is complete before assigned to an approver.
 - [x] All new methods or updated methods have clear docstrings
 - [x] Testing added or updated for new methods
 - [x] Verify if any changes impact the WebPortal Health Checks
 - [x] Approriate documentation updated
 - [x] Changelog file created

### CODE REVIEW

File: api/routes.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ func (api *API) buildHTTPRoutes() {
 	api.staticRouter.DELETE("/user", api.WithDBSession(api.validate(api.userDELETE)))
 	api.staticRouter.GET("/user/limits", api.noValidate(api.userLimitsGET))
 	api.staticRouter.GET("/user/stats", api.validate(api.userStatsGET))
+	api.staticRouter.GET("/user/addpubkey", api.validate(api.userAddPubKeyGET))
+	api.staticRouter.POST("/user/addpubkey", api.validate(api.userAddPubKeyPOST))
```
Comment (ro-tex) [inline api/routes.go:38]: I don't really like the endpoint but `GET /user/pubkey/add` felt wrong. Also `GET /user/pubkey` would be very confusing if it doesn't actually give you the pubkey of the user and returns a challenge instead. So, I'm open to suggestions. Keeping the current name is also fine.

File: Makefile [hunk 2]
```diff
@@ -75,7 +75,7 @@ start-mongo:
      -v $(shell pwd)/test/fixtures/mongo_keyfile:/data/mgkey \
 	mongo:4.4.1 mongod --port=17017 --replSet=skynet --keyFile=/data/mgkey 1>/dev/null 2>&1
 	# wait for mongo to start before we try to configure it
-	sleep 3
+	sleep 4
```
Comment (ChrisSchinnerl) [inline Makefile:78]: Is there no better / more deterministic way to achieve this? e.g. by waiting in a loop for 60 seconds and trying to talk to mongo every second or so?

File: api/handlers.go [hunk 3]
```diff
@@ -546,21 +548,125 @@ func (api *API) userPUT(w http.ResponseWriter, req *http.Request, _ httprouter.P
...
+		api.staticLogger.Warnf("Potential attempt to modify another user's pubKey. Sub of challenge requester '%s', sub of response submitter '%s'", uu.Sub, sub)
+		api.WriteError(w, errors.New("user's sub doesn't match update sub"), http.StatusBadRequest)
+		return
+	}
+	u, err := api.staticDB.UserBySub(ctx, sub, false)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u.PubKeys = append(u.PubKeys, pk)
```
Comment (ChrisSchinnerl) [inline api/handlers.go:656]: Do we want to allow multiple pubkeys per user?

File: api/routes.go (hunk 1 above)
Comment (ChrisSchinnerl) [reply api/routes.go:38]: Why not `/user/pubkeys/register` ? Or just the singular?
Doesn't feel that wrong from a filesystem hierarchy standpoint.

File: test/database/challenge_test.go [hunk 4]
```diff
@@ -112,3 +122,52 @@ func TestValidateChallengeResponse(t *testing.T) {
 		t.Fatal("Failed to create second challenge", err)
 	}
 }
+
+// TestUnconfirmedUserUpdate ensures the entire flow for unconfirmed user
+// updates works as expected.
+func TestUnconfirmedUserUpdate(t *testing.T) {
+	ctx := context.Background()
+	dbName := strings.ReplaceAll(t.Name(), "/", "_")
```
Comment (ChrisSchinnerl) [inline test/database/challenge_test.go:130]: This should be a helper. e.g. `dbNameForTest(t.Name())`

File: api/handlers.go (hunk 3 above)
Comment (ro-tex) [reply api/handlers.go:656]: Yes, that was a specifically requested feature.

File: Makefile (hunk 2 above)
Comment (ro-tex) [reply Makefile]: I was putting this off because I knew it would take some time but... yeah, it had to be done. :)

**Review by m-cat — APPROVED** [review]

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**